### PR TITLE
Add minutes function

### PR DESCRIPTION
**Pull request by JuanMGN**
# Description

Add minutes functions according specification:
https://www.w3.org/TR/sparql11-query/#func-minutes

## Type of change

Make sure you add the correct label for the PR:

- `enhacement` if this PR adds a new feature
- `bug` if it fixes a bug
- `documentation` if it adds docs
- `breaking-change` if this PR introduces a breaking change
- `dependency-updates` if it updates dependencies

### CODE REVIEW

File: modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala [hunk 1]
```diff
@@ -17,8 +18,8 @@ import com.gsk.kg.engine.functions.Literals.nullLiteral
 
 object FuncDates {
 
-  val dateTimeRegex: String =
-    "[0-9]{1,4}-[0-9]{1,2}-[0-9]{1,2}T[0-9]{1,2}:[0-9]{1,2}:[0-9]{1,2}"
+  private val Hours   = 3
```
Comment (patofp) [inline modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala:21]: Why are these defined globally? I can only see them used on their corresponding function scope.

File: modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala [hunk 2]
```diff
@@ -101,4 +97,28 @@ object FuncDates {
...
+  /** Get hours, minutes of dateTime column
+    * @param col
+    * @param pos
+    * @return Column with
+    *         Integer if hours or minutes
+    */
+  private def getTimeFromDateTimeCol(col: Column, pos: Int): Column = {
+    val dateTimeRegex: String =
+      "[0-9]{1,4}-[0-9]{1,2}-[0-9]{1,2}T[0-9]{1,2}:[0-9]{1,2}:[0-9]{1,2}"
+    val len = 2
```
Comment (patofp) [inline modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala:110]: Where is this variable being used?

**Review by patofp — CHANGES_REQUESTED** [review]
Just a couple of things, for the rest LGTM!

File: modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala (hunk 2 above)
Comment (JuanMGN) [reply modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala:110]: len is not used, I forgot delete it

File: modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala (hunk 1 above)
Comment (JuanMGN) [reply modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala:21]: These variables( Minutes and Hours) are used when we call to function getTimeFromDateTimeCol. I think the code is more readable if the numbers are declared as a variable

Comment (patofp) [reply modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala]: Yeah, it is the scope of the variable what it should be changed. 😉

**Review by patofp — APPROVED** [review]
LGTM!

File: modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala (hunk 1 above)
Comment (JuanMGN) [reply modules/engine/src/main/scala/com/gsk/kg/engine/functions/FuncDates.scala]: done :)

**Pull request merged**